docs/product-assembly_migrating-from-BRMS-6.4-to-DM-7.0/src/main/asciidoc/code-troubleshooting-ref.adoc

```diff
@@ -1,58 +0,0 @@
-** `org.kie.api.task.*`
-** `org.kie.api.executor.*`
-** `org.kie.api.concurrent.*`
-** `org.kie.api.builder.*`
-** `org.kie.api.command.*`
-** `org.kie.api.runtime.*`
-* `drools-core`: the rules engine. The changed classes are:
-** `org.drools.core.command.*`
-** `org.drools.core.common.*`
-* `kie-server-api`: the general API, including commands,model and so on, for KIE Server ({KIE_SERVER}). The changed classes are:
-** `org.kie.server.api.commands.*`
-** `org.kie.server.api.marshalling.*`
-** `org.kie.server.api.model.*`
-** `org.kie.server.api.rest.RestURI` (constants have slightly changed, omitting leading `/`)
-* `kie-server-controller-api`: the API for the KIE Server Controller. The changed classes are:
-** `org.kie.server.controller.api.service.*`
-* `kie-server-controller-rest`: the REST API for KIE Server Controller. The changed classes are:
-** `org.kie.server.controller.rest.RestSpecManagementServiceImpl`
-** `org.kie.server.controller.rest.RestKieServerControllerImpl`
-* `kie-server-client`: the KIE Server Client. The changed classes are:
-** `org.kie.server.client.SolverServicesClient`
-** `org.kie.server.client.UIServicesClient`
-** `org.kie.server.client.admin.ProcessAdminServicesClient`
-** `org.kie.server.client.ProcessServicesClient`
-** `org.kie.server.client.QueryServicesClient`
-** `org.kie.server.client.JobServicesClient`
-** `org.kie.server.client.UserTaskServicesClient`
-** `org.kie.server.client.KieServicesClient`
-** `org.kie.server.client.KieServicesConfiguration`
-
-If you encounter build errors with any of the listed classes, review the detailed reports that are available at https://access.redhat.com/articles/3352151 . The reports indicate the API changes. You can use this information to adapt your code to the changes.
-
-== Logic changes in Drools modules
-
-The following logic changes were made in Drools modules:
-
-* In Red Hat BRMS 6.4, when a rule executes the sum function in an accumulate pattern, the result always returns as a double data type regardless of the data type of the inputs. {PRODUCT} {PRODUCT_VERSION} preserves the data type of the inputs on which the sum is executed. This enhancement provides a more accurate result from the sum function. In the following example, the result type of the accumulate function is `Long` instead of `double`:
-+
-[source,java]
-----
-Long(...) from accumulate(..., sum($p.getLongWeight()))
-----
-+
-If the rules in your Red Hat BRMS 6.4 project include the sum function in an accumulate pattern, search for these functions and review them. Keep in mind that these functions return a double data type in Red Hat BRMS 6.4, but will return the data type of the input values in {PRODUCT} {PRODUCT_VERSION}.
-* When no fact matches the accumulate pattern, min and max accumulate functions in {PRODUCT} {PRODUCT_VERSION} do not return `+/-Integer.MAX_VALUE`, but return `null`, therefore, unlike BRMS 6.4, the accumulate in the rule is not matched and the rule does not fire.
```

Comment:
..accumulate in the rule... should this be accumulate pattern in the rule?

Reply:
Accumulate is a DRL fact/class.